### PR TITLE
Feed promotions to renew [wip]

### PR DESCRIPTION
**Pull request by AWare**
Also contains a not great attempt at passing the country through, but I really don't like it as it's no longer using a gu.Country and some of the direct-debit logic is less clear than before so it needs tidying after christmas.

### CODE REVIEW

**Comment by AWare** [comment]
mc: https://github.com/guardian/membership-common/pull/388

File: app/controllers/AccountManagement.scala [hunk 1]
```diff
@@ -249,18 +249,18 @@ object AccountManagement extends Controller with LazyLogging with CatalogProvide
...
         }
         case _: Product.Weekly => subscription.asWeekly.map { weeklySubscription =>
-          ManageWeekly(billingSchedule, weeklySubscription)
+          ManageWeekly(billingSchedule, weeklySubscription, promoCode)
         }
       }
       maybeFutureManagePage.getOrElse {
         // the product type didn't have the right charges
-        Future.successful(Ok(views.html.account.details(None)))
+        Future.successful(Ok(views.html.account.details(None,promoCode)))
```
Comment (johnduffell) [inline app/controllers/AccountManagement.scala]: watch your spacing (maybe you plan to reformat later?)

File: app/controllers/Promotion.scala [hunk 2]
```diff
@@ -36,38 +36,45 @@ object Promotion extends Controller with LazyLogging with CatalogProvider {
...
-      val body = Json.obj(
-        "promotion" -> Json.toJson(promo),
-        "adjustedRatePlans" -> Json.toJson(getAdjustedRatePlans(promo,country)),
-        "isValid" -> result.isRight,
-        "errorMessage" -> result.swap.toOption.map(_.msg)
-      )
-      result.fold(_ => NotAcceptable(body), _ => Ok(body))
+    if (tpBackend.promoService.isEmpty) {
+      logger.error("No promotions loaded.")
+      InternalServerError("")
```
Comment (johnduffell) [inline app/controllers/Promotion.scala]: see comment on https://github.com/guardian/membership-common/pull/388#discussion-diff-94567788R16

**Comment by AWare** [comment]
https://github.com/guardian/membership-common/pull/396

**Pull request merged**